### PR TITLE
Quiz listing page

### PR DESCRIPTION
**Pull request by vishal804**
Hello 

### Functionality 

 Quiz listing and Rule Page
 Question and Result Page in quiz
 
 ### Files to review

src/component/questionCard/QuestionCard.js
src/pages/quizListingPage/QuizListingPage.js
src/pages/quizPage/QuizPage.js
src/pages/resultPage/ResultPage.js
src/pages/rulePage/RulePage.js

### CODE REVIEW

**Review by Balakumar3018 — APPROVED** [review]
Functionality working as expected. Good to merge

**Review by Himanshu-Dhawale — APPROVED** [review]
1. UI looks good
2. Functionalities are also working
3. Code also looks clean

File: src/component/questionCard/QuestionCard.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        const response = await axios.get(
+          `/api/quizzes/${quizId}/${questionId}`,
+          {
+            headers: { authorization: token },
+          }
+        );
+        quizDispatch({ type: "SET_QUESTION", payload: response.data.question });
+      } catch (error) {
+        Navigate("/");
+        console.log(error);
```
Comment (irajbhinde) [inline src/component/questionCard/QuestionCard.js]: You can remove the console log here

File: src/component/questionCard/QuestionCard.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                </button>
+              </Link>
+            )}
+          </div>
+        </>
+      )}
+    </>
+  );
+};
+
```
Comment (irajbhinde) [inline src/component/questionCard/QuestionCard.js:102]: Recently I learned a way to make our code inside return look clean. What you can do is, before the return statement where we write all the logic you can make components. For example - 

export default func XYZ(){

const yourComponent = ( 

write all your jsx here

)

return (

yourComponent


)
}

File: src/pages/quizPage/QuizPage.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from "react";
+import "./quizPage.css";
+import axios from "axios";
+import { QuestionCard } from "../../component";
+import { useQuiz } from "../../context/quiz-context";
```
Comment (irajbhinde) [inline src/pages/quizPage/QuizPage.js:5]: You can create index.js for all the context & then import them in single line here. This is fine as well though if you have less contexts

**Review by irajbhinde — COMMENTED** [review]
Looks good, kindly look into the comment I gave for the code inside return statement, you can try it in your future project / files

**Comment by irajbhinde** [comment]
Your deploy preview doesn't seem to be working. Kindly look into it

**Review by manojasarsa — APPROVED** [review]
The code seems clean. 
Functionality is working as expected.

File: src/component/questionCard/QuestionCard.js (hunk 2 above)
Comment (vishal804) [reply src/component/questionCard/QuestionCard.js:102]: will try that

File: src/pages/quizPage/QuizPage.js (hunk 3 above)
Comment (vishal804) [reply src/pages/quizPage/QuizPage.js:5]: There are not many contexts here

File: src/component/questionCard/QuestionCard.js (hunk 1 above)
Comment (vishal804) [reply src/component/questionCard/QuestionCard.js]: Done the changes

**Pull request merged**